### ISSUE TITLE
(t) Update Middlewares style

### ISSUE THREAD

**Issue by FroggyFlox** [rockstor/rockstor-core#2662]
Thanks to @phillxnet, we now have a config to run with the Python's warning flag (#2655).
One of these warnings is relates to our use of Middlewares:
```
RemovedInDjango20Warning: Old-style middleware using settings.MIDDLEWARE_CLASSES is deprecated. Update your middleware and use settings.MIDDLEWARE instead.
  "instead.", RemovedInDjango20Warning
```

This most likely relates to the deprecation of `MIDDLEWARE_CLASSES` in Django 1.10:
https://docs.djangoproject.com/en/1.11/ref/settings/#middleware-classes

> Deprecated since version 1.10: Old-style middleware that uses settings.MIDDLEWARE_CLASSES are deprecated. [Adapt old, custom middleware](https://docs.djangoproject.com/en/1.11/topics/http/middleware/#upgrading-middleware) and use the [MIDDLEWARE](https://docs.djangoproject.com/en/1.11/ref/settings/#std:setting-MIDDLEWARE) setting.
> This was the default setting used in Django 1.9 and earlier. Django 1.10 introduced a new style of middleware. If you have an older project using this setting you should [update any middleware you’ve written yourself](https://docs.djangoproject.com/en/1.11/topics/http/middleware/#upgrading-middleware) to the new style and then use the [MIDDLEWARE](https://docs.djangoproject.com/en/1.11/ref/settings/#std:setting-MIDDLEWARE) setting.

Note that this was fully removed from Django 2.0:
https://docs.djangoproject.com/en/4.2/releases/2.0/#features-removed-in-2-0
> Support for old-style middleware using settings.MIDDLEWARE_CLASSES is removed.

**Comment by FroggyFlox** [comment]
We currently have the following:
https://github.com/rockstor/rockstor-core/blob/3f8794231e2135201c4587bc4d8b027b621f258a/src/rockstor/settings.py#L148-L158

We thus have:
- several Django's included middlewares
- 1 custom middleware

Note the following from  Django's documentation:
https://docs.djangoproject.com/en/1.11/topics/http/middleware/#upgrading-pre-django-1-10-style-middleware
> Django provides django.utils.deprecation.MiddlewareMixin to ease creating middleware classes that are compatible with both [MIDDLEWARE](https://docs.djangoproject.com/en/1.11/ref/settings/#std:setting-MIDDLEWARE) and the old [MIDDLEWARE_CLASSES](https://docs.djangoproject.com/en/1.11/ref/settings/#std:setting-MIDDLEWARE_CLASSES). All middleware classes included with Django are compatible with both settings.

We should thus be OK for the middlewares included in Django that we use.
We would need to adjust our custom middleware, however.

**Comment by phillxnet** [comment]
Closing as:
Fixed by #2663
